include/h2o.h

```diff
@@ -503,6 +558,11 @@ struct st_h2o_context_t {
         h2o_timestamp_string_t *value;
     } _timestamp_cache;
 
+    /**
+     * counter for errors internally emitted by h2o
+     */
+    h2o_emitted_errors_t emitted_errors;
```

Comment:
Thank you for the changes.  The commit looks good.

Some additional comments from me are as follows.  How do they sound?

1. instead of updating the counters in `h2o_send_error_deferred`, rename the function to `h2o_send_error_generic_deferred` that does not do the update, and instead modify the counters at the location that calls the deferred function
2. with 1 being done, we can remove h2o_context_report_http1_error
3.  remove E_HTTP4xx or E_HTTP5xx now that we have all the error codes that updates the counter hard-coded